### ISSUE TITLE
Decode function call

### ISSUE THREAD

**Issue by jgeary** [ethereum/web3.js#3016]
### Description 
I cannot figure out how to decode the result of `web3.eth.abi.encodeFunctionCall(...)` to get the original parameters. Not sure if this is due to lacking functionality or if this is possible but not yet documented.

#### Expected behavior
After calling
```
const result = web3.eth.abi.encodeFunctionCall({
    name: 'myMethod',
    type: 'function',
    inputs: [{
        type: 'uint256',
        name: 'myNumber'
    },{
        type: 'string',
        name: 'myString'
    }]
}, ['2345675643', 'Hello!%']);
```
and getting the result `"0x24ee0097000000000000000000000000000000000000000000000000000000008bd02b7b0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000748656c6c6f212500000000000000000000000000000000000000000000000000"`,
I want to call `web3.eth.abi.decodeParameters(['bytes4', 'uint256', 'string'], result)` to get an array which includes `'2345675643'` and `'Hello!%'`. 

#### Actual behavior

`'Uncaught Error: overflow (operation="setValue", fault="overflow", details="Number can only safely store up to 53 bits")'`

### Versions 
* web3.js: 1.2.1
* nodejs: 11.15.0
* browser: Chrome

**Comment by nivida** [comment]
This error occurs because you're calling the ``decodeParameters`` function with the wrong types array. The correct types array would be: ``['uint256', 'string']``. 

Example: 

``` javascript 
const result = web3.eth.abi.encodeFunctionCall({
    name: 'myMethod',
    type: 'function',
    inputs: [{
        type: 'uint256',
        name: 'myNumber'
    },{
        type: 'string',
        name: 'myString'
    }]
}, ['2345675643', 'Hello!%']);

web3.eth.abi.decodeParameters(['uint256', 'string'], result)
```

**Comment by jgeary** [comment]
@nivida thanks for getting back to me. the code you provided still throws the same exact error.

**Comment by sterlu** [comment]
Hey I just came across a similar error half-asleep last night. I figured it out this morning so I'll share the answer in case someone needs it. @jgeary was encoding a *function call* and decoding only the *argument*. Function calls are composed by the function ID (first 4 bytes/8 characters) and then the concatenated parameters. 

TLDR just slicing off the first 8 chars from the call data should do what you're looking for.
```js
_web3.eth.abi.decodeParameters(['uint256', 'string'], '0x000000000000000000000000000000000000000000000000000000008bd02b7b0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000748656c6c6f212500000000000000000000000000000000000000000000000000')
// > {0: "2345675643", 1: "Hello!%" }
```

**Comment by miljantekic** [comment]
Can confirm that what @sterlu wrote is the way to go. Just slice the function signature out of the transaction input.

**Comment by dzimiks** [comment]
Big thanks @sterlu!!! 😎

**Comment by defido** [comment]
[sterlu](https://github.com/sterlu)
@sterlu 
I love you with every part of my body right now. This solved this issue for me. MY GOD! Why is this not in the web3js docs!?

For others, this is an ugly solution, but it works: (There is 1000 better ways to remove characters from a string in JS)

```
let web3 = new Web3('https://bsc-dataseed.binance.org/');
		const a = await web3.eth.getTransaction('0x041e65ab1b3ef192d1e2ae38d167f6293f026b20da980f4987e173c9ffcfc886');
		let find = a.input.slice(2, 10);
		let remove = a.input.replace(find, '');
		const z = await web3.eth.abi.decodeParameters(
			[
				{
					type: 'address',
					name: 'token',
				},
				{
					type: 'uint256',
					name: 'amount',
				},
				{
					type: 'address',
					name: 'receiverAddress',
				},
				{
					type: 'uint256',
					name: 'targetChainId',
				}
			],
			remove <--- The input with the function ID removed.
		);
```
OPTION 2:
```
let web3 = new Web3('https://bsc-dataseed.binance.org/');
		const a = await web3.eth.getTransaction('0x041e65ab1b3ef192d1e2ae38d167f6293f026b20da980f4987e173c9ffcfc886');
let remove = `0x${a.input.substr(10)}`; <---- Remove the function ID, add back in the 0x
		const z = await web3.eth.abi.decodeParameters(
			[
				{
					type: 'address',
					name: 'token',
				},
				{
					type: 'uint256',
					name: 'amount',
				},
				{
					type: 'address',
					name: 'receiverAddress',
				},
				{
					type: 'uint256',
					name: 'targetChainId',
				}
			],
			remove
		);

```
